### PR TITLE
clean up package.json

### PR DESCRIPTION
**Pull request by briangough**
### Description

Clean up the package.json file, 

- move modules used by tests into devDependencies
- remove unused lynx module
- update `request` to latest version

#### Screenshots

NA

#### Related Issues / PRs

NA

### Review

Small changes.

#### Potential Impact

Low

#### Manual Testing Performed

- [X]  Acceptance tests

#### Accessibility

NA

### Deployment

NA

#### Deployment Checklist

NA

#### Metrics and Monitoring

NA

#### Who Needs to Know?

NA

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,22 +25,21 @@
     "express": "3.11.0",
     "lodash": "^4.17.13",
     "logger-sharelatex": "^1.7.0",
-    "lynx": "0.0.11",
     "metrics-sharelatex": "^2.2.0",
     "mongojs": "^2.6.0",
     "redis-sharelatex": "^1.0.11",
-    "request": "2.47.0",
+    "request": "^2.47.0",
     "requestretry": "^1.12.0",
```
Comment (das7pad) [inline package.json]: This package pulls in an old version of the `request` package - essentially ignoring your bump.

See the diff from moving to 3.x https://github.com/das7pad/document-updater-sharelatex/commit/a66626c03deef96278a309747220fb5fc71f41b9

From the change log https://github.com/FGRibreau/node-request-retry/blob/master/CHANGELOG.md:
- 2.0 `request` as `peerDependecy`
- 3.0 retry on status code 429 
- 4.0 signature change for `retryStrategy` (which we do not use)

**Comment by briangough** [comment]
Thanks, I've updated requestretry.  I tried to remove bunyan but the service would not start locally after that, so I had to keep it.

**Comment by das7pad** [comment]
https://github.com/googleapis/nodejs-logging-bunyan/blob/f5f36beedfd4eac65e8e035c3266c68d2cd35ef4/package.json#L90

`@google-cloud/logging-bunyan` has a peer dependency set on `bunyan`. But I would expect that npm links the `bunyan` package of the logger module then.

**Comment by briangough** [comment]
> https://github.com/googleapis/nodejs-logging-bunyan/blob/f5f36beedfd4eac65e8e035c3266c68d2cd35ef4/package.json#L90
> 
> `@google-cloud/logging-bunyan` has a peer dependency set on `bunyan`. But I would expect that npm links the `bunyan` package of the logger module then.

Yes, I thought it would be fine but somehow it didn't work locally - even after removing node_modules and reinstalling everything (it did work in cloud build, so something funny is going on).

**Comment by das7pad** [comment]
Let's move `bunyan` to the regular dependencies then -- given that it's used in production by the `logger-sharelatex` -> `@google-cloud/logging-bunyan`

**Pull request merged**